src/main/pileup.js

```diff
@@ -80,6 +80,37 @@ function create(elOrId: string|Element, params: PileupParams): Pileup {
       ReactDOM.render(<Root referenceSource={referenceTrack.source}
                             tracks={vizTracks}
                             initialRange={params.range} />, el);
+
+  //if the element doesn't belong to document DOM observe DOM to detect
+  //when it's attached
+  var observer = null;
+  if (!document.contains(el)) {
+    observer = new MutationObserver(function(mutations) {
+      mutations.forEach(function(mutation) {
+        if (mutation.type === 'childList') {
+          var added= false;
+          for (var i=0; i<mutation.addedNodes.length;i++) {
+            if (mutation.addedNodes[i]===el) {
+              added = true;
+            }
+          }
+          if (added) {
+            if (reactElement) {
+              reactElement.setState({updateSize:true});
+            } else {
+              throw 'ReactElement was not initialized properly';
```

Comment:
We can move it up. But it must be inside MutationObserver constructor.
There might be some issues due to lifecycle of react component (I'm not familiar with it at all): 
* consider object creation
* object destruction - which method will be called when - maybe observer still will be running when react element is gone??

Anyway, where do you want to move it? Inside forEach loop? (line 90)